src/swaps/lifi/LifiSwapProvider.ts

```diff
@@ -0,0 +1,321 @@
+import { HttpClient } from '@chainify/client';
+import { EvmTypes } from '@chainify/evm';
+import { Transaction, TransactionRequest, TxStatus } from '@chainify/types';
+import LiFi, { ChainId, GasCost, LifiStep, Orders, Step } from '@lifi/sdk';
+import { chains, currencyToUnit, unitToCurrency } from '@liquality/cryptoassets';
+import BN from 'bignumber.js';
+import { v4 as uuidv4 } from 'uuid';
+import { ActionContext } from '../../store';
+import { withInterval, withLock } from '../../store/actions/performNextAction/utils';
+import { prettyBalance } from '../../utils/coinFormatter';
+import cryptoassets from '../../utils/cryptoassets';
+import { ChainNetworks } from '../../utils/networks';
+import { EvmSwapHistoryItem, EvmSwapProvider, EvmSwapProviderConfig } from '../EvmSwapProvider';
+import {
+  EstimateFeeRequest,
+  EstimateFeeResponse,
+  NextSwapActionRequest,
+  QuoteRequest,
+  SwapRequest,
+  SwapStatus,
+} from '../types';
+
+export interface LifiSwapProviderConfig extends EvmSwapProviderConfig {
+  rpcURL: string;
+}
+
+export interface LifiSwapHistoryItem extends EvmSwapHistoryItem {
+  fromFundHash: string;
+  fromFundTx: Transaction<EvmTypes.EthersTransactionResponse>;
+}
+
+class LifiSwapProvider extends EvmSwapProvider {
+  readonly config: LifiSwapProviderConfig;
```

Comment:
good usage of readonly, can we also specify the visibility modifier? same for the rest of the class variables